.github/workflows/preview-docs.yml

```diff
@@ -0,0 +1,162 @@
+
+            - name: Build Storybook
+              run: yarn storybook:build
+
+            ## --- DEPLOY TO AZURE BLOB STORAGE --- ##
+            - name: Install AzCopy
+              run: |
+                  # Download and install AzCopy
+                  wget -O azcopy.tar.gz https://github.com/Azure/azure-storage-azcopy/releases/download/v10.29.1/azcopy_linux_amd64_10.29.1.tar.gz
+                  tar -xf azcopy.tar.gz --strip-components=1
+                  sudo mv azcopy /usr/local/bin/
+                  azcopy --version
+
+            - name: Deploy to Azure Blob Storage
+              id: deploy
+              env:
+                  AZURE_STORAGE_SAS_TOKEN: ${{ secrets.AZURE_STORAGE_SAS_TOKEN }}
+                  PR_HASH: ${{ steps.pr_hash.outputs.hash }}
+              run: |
+                  # Check if AZURE_STORAGE_SAS_TOKEN is set
+                  if [ -z "${AZURE_STORAGE_SAS_TOKEN}" ]; then
+                      echo "Missing Azure Storage SAS Token"
+                      exit 1
+                  fi
+
+                  # Disable debug mode to prevent token leakage
+                  set +x
+
+                  # Clean the SAS token (remove any whitespace/newlines)
+                  CLEAN_SAS_TOKEN=$(echo "${AZURE_STORAGE_SAS_TOKEN}" | tr -d '\n\r\t ')
+
+                  # Upload documentation
+                  echo "Uploading documentation to ${PR_HASH}/docs/"
+                  azcopy copy "projects/documentation/dist/*" \
+                    "https://swcpreviews.blob.core.windows.net/\$web/${PR_HASH}/docs/?${CLEAN_SAS_TOKEN}" \
+                    --recursive \
+                    --from-to LocalBlob
+
+                  # Upload Storybook
+                  echo "Uploading Storybook to ${PR_HASH}/docs/storybook/"
+                  azcopy copy "storybook-static/*" \
+                    "https://swcpreviews.blob.core.windows.net/\$web/${PR_HASH}/docs/storybook/?${CLEAN_SAS_TOKEN}" \
+                    --recursive \
+                    --from-to LocalBlob
+
+                  # Set output URLs
+                  docs_url="https://swcpreviews.z13.web.core.windows.net/${PR_HASH}/docs/"
+                  storybook_url="https://swcpreviews.z13.web.core.windows.net/${PR_HASH}/docs/storybook/"
+                  echo "docs_url=${docs_url}" >> $GITHUB_OUTPUT
+                  echo "storybook_url=${storybook_url}" >> $GITHUB_OUTPUT
+                  echo "Deployed to: ${docs_url}"
+
+    close_pull_request_job:
+        if: github.event_name == 'pull_request' && github.event.action == 'closed'
+        runs-on: ubuntu-latest
+        name: Clean up PR deployment
+        steps:
+            - name: Generate PR hash
+              id: pr_hash
+              run: |
+                  # Create the same hash as in the deploy job
+                  pr_hash="pr-${{ github.event.pull_request.number }}"
+                  echo "hash=${pr_hash}" >> $GITHUB_OUTPUT
+
+            - name: Install AzCopy
```

Comment:
I see we're also installing AzCopy for this job, could we make this centralized and re-use in both places?